manifests/osfamily/windows.pp

```diff
@@ -0,0 +1,37 @@
+class puppet_agent::osfamily::windows {
+
+  $_arch = $::kernelmajversion ?{
+    /^5\.\d+/ => 'x86', # x64 is never allowed on windows 2003
+    default   => $::puppet_agent::arch,
+  }
+
+  $_source = $::puppet_agent::source ? {
+    undef   => "https://downloads.puppetlabs.com/windows/puppet-agent-${_arch}-latest.msi",
```

Comment:
b/c we need to build the default based on Architecture and we won't know what arch they have passed or if they are capable of installing.